src/pkg/packager/composer/list.go

```diff
@@ -22,17 +22,48 @@ import (
 type Node struct {
 	types.ZarfComponent
 
+	index int
+
 	vars   []types.ZarfPackageVariable
 	consts []types.ZarfPackageConstant
 
-	relativeToHead string
+	relativeToHead      string
+	importURL           string
+	originalPackageName string
 
 	prev *Node
 	next *Node
 }
 
+// GetIndex returns the .components index location for this node's source `zarf.yaml`
+func (n *Node) GetIndex() int {
+	return n.index
+}
+
+// GetOriginalPackageName returns the .metadata.name of the zarf package the component originated from
+func (n *Node) GetOriginalPackageName() string {
+	return n.originalPackageName
+}
+
+// ImportLocation gets the path from the base zarf file to the imported zarf file
+func (n *Node) ImportLocation() string {
+	if n.importURL != "" {
+		return n.importURL
+	}
+	return n.relativeToHead
+}
```

Comment:
what added benefit is keeping track of `n.importURL` vs just looking at `n.Import.URL`?